src/main/java/edu/cornell/kfs/ksr/document/validation/impl/SecurityGroupRule.java

```diff
@@ -131,29 +129,18 @@ public boolean validateSecurityGroupTabs(MaintenanceDocument document) {
 		boolean successTabName = true;
 
 		List<SecurityGroupTab> securityGroupTabs = ((SecurityGroup) document.getDocumentDataObject()).getSecurityGroupTabs();
-		if (securityGroupTabs != null) {
+		if (CollectionUtils.isNotEmpty(securityGroupTabs)) {
 			for (int i = 0; i < securityGroupTabs.size(); i++) {
 				SecurityGroupTab tab = securityGroupTabs.get(i);
-
-				if ((tab.getTabOrder() == null) && (tab.getTabName() == null)) {
-					securityGroupTabs.remove(i);
-				}
-				else if (tab.getTabOrder() == null) { // Reset to old value
-					tab.setTabOrder(((SecurityGroup) document.getOldMaintainableObject().getDataObject()).getSecurityGroupTabs().get(i).getTabOrder());
-				}
-				if (tab.getTabName() == null) {
-					tab.setTabName(((SecurityGroup) document.getOldMaintainableObject().getDataObject()).getSecurityGroupTabs().get(i).getTabName());
-				}
-
 				for (int j = i + 1; j < securityGroupTabs.size(); j++) {
 					SecurityGroupTab tempTab = securityGroupTabs.get(j);
-					if (successTabOrder
+					if (successTabOrder && tab.getTabOrder() != null
 							&& tab.getTabOrder().equals(tempTab.getTabOrder())) {
 						GlobalVariables.getMessageMap().putErrorForSectionId(KSRConstants.SECTION_SECURITY_TABS, KSRKeyConstants.ERROR_SECURITY_GROUP_TAB_ORDER_UNIQUE);
 						successTabOrder = false;
 					}
-					if (successTabName
-							&& tab.getTabName().equals(tempTab.getTabName())) {
+					if (successTabName && StringUtils.isNotBlank(tab.getTabName())
+							&& StringUtils.equals(tab.getTabName(), tempTab.getTabName())) {
```

Comment:
Same question with this one too, would string case be an issue?

Reply:
Updated to be case-insensitive.